vendmachine/VendingMachine.java

```diff
@@ -0,0 +1,130 @@
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.io.FileWriter;
+import java.io.FileReader;
+import java.io.BufferedWriter;
+import java.io.BufferedReader;
+import java.io.IOException;
+
+class VendingMachine {
+    private HashMap<String, Product> products = new HashMap<>();
+    private BufferedWriter log = null;
+    private String user = null;
+
+    private void close() {
+        try {
+            BufferedWriter state = new BufferedWriter(new FileWriter("state.csv"));
+            Iterator it = products.values().iterator();
+            while (it.hasNext()) {
+                Product p = (Product) it.next();
+                state.write(p.getName()+", "+p.getPrice()+", "+p.getDescription()+"\n");
+            }
+            state.close();
+            log.close();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    private void log(String message) {
+        if (log == null) 
+            try {
+                log = new BufferedWriter(new FileWriter("transactions.log", true));
+            } catch (IOException e) {
+                e.printStackTrace();
```

Comment:
💡  I would rather throw an exception here. Please consider also the `throw` best practice from above.